### ISSUE TITLE
Remove rotating credit card brand logos from checkout page

### ISSUE THREAD

**Issue by ricardo** [woocommerce/woocommerce-gateway-stripe#2033]
Internal discussion: p1633447195109500/1633383549.107300-slack-CHG7MTCAF

Some merchants [complained](https://wordpress.org/support/topic/remove-animation-for-stripe-plugin-revert-to-previous-version/) about the recent change in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1979, so we want to revert that and remove the credit card icons altogether.

**Comment by jkohlbach** [comment]
Hi @ricardo "and remove the credit card icons altogether." was going too far ...

From the original complaint thread people just didn't appreciate the switcharoo of going from static icons to animated without having a say in it or an option to revert. It caused checkout display issues.

Can you please re-instate the static icons as they were?

**Comment by pjv** [comment]
>Can you please re-instate the static icons as they were?

Yes please. Static icons were not broken, didn’t need to be fixed.

**Comment by simonkoehler** [comment]
Same here. Bring them back. They are important for the shops conversion rate.

**Comment by jkohlbach** [comment]
**Another reason:** It might also be breaking terms with payment processors to not be displaying the icons as someone pointed out in this thread: https://wordpress.org/support/topic/static-icons-not-showing-on-5-6-2/

I've also added code there to re-add in case any store owners stumble across this ticket.

**Comment by ricardo** [comment]
Also copying my response from the forum thread here:

Thanks for the feedback and apologies for the inconvenient change.

This change was mainly introduced because of the (early access) new checkout experience, which can offer multiple Stripe payment methods under the same payment gateway option on the checkout page.

Other than the inconsistency with that, we also had another issue where the icons would appear misaligned in different templates, because of the amount of icons, hence the first change where we introduced the rotating credit card brands, which were consistent with the new checkout experience.

We’re taking your feedback into consideration for the next releases.

For now, I would recommend @jkohlbach’s alternative to bring the icons back. Here’s a snippet for all the credit card brands:

```php
function add_credit_card_gateway_icons( $icon_string, $gateway_id ) {
	if ( 'stripe' === $gateway_id ) {
		$icon_string  = '<img src="' . WC_STRIPE_PLUGIN_URL . '/assets/images/visa.svg" class="stripe-visa-icon stripe-icon" alt="Visa" />';
		$icon_string .= '<img src="' . WC_STRIPE_PLUGIN_URL . '/assets/images/mastercard.svg" class="stripe-mastercard-icon stripe-icon" alt="Mastercard" />';
		$icon_string .= '<img src="' . WC_STRIPE_PLUGIN_URL . '/assets/images/amex.svg" class="stripe-amex-icon stripe-icon" alt="American Express" />';
		$icon_string .= '<img src="' . WC_STRIPE_PLUGIN_URL . '/assets/images/discover.svg" class="stripe-discover-icon stripe-icon" alt="Discover" />';
		$icon_string .= '<img src="' . WC_STRIPE_PLUGIN_URL . '/assets/images/diners.svg" class="stripe-diners-icon stripe-icon" alt="Diners" />';
		$icon_string .= '<img src="' . WC_STRIPE_PLUGIN_URL . '/assets/images/jcb.svg" class="stripe-jcb-icon stripe-icon" alt="JCB" />';
	}
	return $icon_string;
}

add_filter( 'woocommerce_gateway_icon', 'add_credit_card_gateway_icons', 10, 2 );
```

**Comment by Hamody78** [comment]
Hi
Please, I copied the code and paste it in css option on customization page but it did not work,
Please tell me, where I could put this code?
thank you

**Comment by PPjev** [comment]
> Hi Please, I copied the code and paste it in css option on customization page but it did not work, Please tell me, where I could put this code? thank you

functions.php file of your theme

**Comment by iguanamom** [comment]
Can someone help me out here? I apologize in advance for being a coding idiot. I've used that workaround in my functions.php and after fiddling with it to get it right, replacing ' . WC_STRIPE_PLUGIN_URL . ' with the actual URL and testing that URL to be sure it actually goes there (it does), no cards are showing on checkout. I thought maybe it's because I'm in test mode for the credit cards...nope. I've cleared cache on every change and they simply don't show up. I assume the single quotes need to be removed as that crashed the site when I had them in there. 

Then I thought maybe it has something to do with USD vs. NZD even though it's not mentioned in this snippet but no. So as an example, the first few lines of the code I have are (removing the actual website and changing it to MYWEBSITE):

function add_credit_card_gateway_icons( $icon_string, $gateway_id ) {
	if ( 'stripe' === $gateway_id ) {
		$icon_string  = '<img src="https://MYWEBSITE/wp-content/plugins/woocommerce-payments/assets/images/cards/visa.svg" class="stripe-visa-icon stripe-icon" alt="Visa" />';

(*When I published this, it hid the URL so I don't know if you'll be able to see it!? In case you can't I'll shorten it:)
https://MYWEBSITE/wp-content/plugins/woocommerce-payments/assets/images/cards/visa.svg" class="stripe-visa-icon stripe-icon" alt="Visa"

and the rest follow that format replacing the end with the right card svg. The original snip provided didn't have the 'cards' folder, but mine does. Only other thing is that the icons are actually really large. Do they need to be icon sized or does the code make them icon size? That's all I can think of that's stopping them from showing up.

Thanking you in advance!

**Comment by ricardo** [comment]
@iguanamom, I noticed the URL you mentioned is not quite right. There's no `/cards/` folder.

If you add the code snippet [above](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2033#issuecomment-939216545) as is into the theme functions.php file (**Appearance > Theme Editor > functions.php**), it should bring back the previous behavior.

No need to change `WC_STRIPE_PLUGIN_URL`, as this is a constant for the Stripe plugin folder URL.

Unless you have any issues with your theme, it shouldn't change the sizing of the SVG files.

**Comment by iguanamom** [comment]
@ricardo thank you for your help here! Figures, I'm trying to make things more complicated. HOWEVER, this still didn't work for me. I can definitely confirm I have a 'cards' folder where the images are stored, but even putting that in there (without the full URL as I was doing before, just adding it onto the path) it still won't show up. So trying both ways, as written and adding in the 'cards' folder, nothing. I've cleared caches between each attempt. I'm definitely editing the right file. 

I'm using OceanWP which does have its own Woo styling so I disabled that and tried again, but still won't show. I also disabled the testing option for the credit cards and that didn't make a difference either. 

I'd assume the path definitely should have the 'cards' folder in there, shouldn't it? That's where the images are stored.
wp-content/plugins/woocommerce-payments/assets/images/cards  is the path I show for my install.

I decided to switch to another theme to see if that made a difference and it did not, that one also did not show the icons.

Well, I can understand if you are unable to chime in any further here, but figured I'd report back. Thank you for your time and effort!

**Comment by ricardo** [comment]
@iguanamom I just noticed another weird thing in your URL: `woocommerce-payments` 😅.

This repository is for a different plugin (`woocommerce-gateway-stripe`), so the code snippet I sent you above will ultimately only work for the Stripe plugin, not WooCommerce Payments.

WooCommerce Payments doesn't seem to provide a way to customize the gateway icon, so I'm afraid you can't do that with WooCommerce Payments.

**Comment by iguanamom** [comment]
@ricardo ahhhhh ok, thanks for noticing that. Dumb there are two very similar plugins for this. I wonder what the actual difference is!? I'll get on the plugin support page and ask there. Thanks again for your time!!

**Comment by ricardo** [comment]
@iguanamom Both plugins use Stripe for payment processing and both plugins offer an extensive set of features.

There are small changes in terms of features, fees and supported countries, but the main difference is that WooCommerce Payments offers a more streamlined approach to start collecting payments - You don't need to configure API keys or do pretty much anything via the Stripe dashboard, while in the Stripe payment gateway you have your separate standard account and the integration is more manual, it's not better or worse, it's simply different.

**Comment by iguanamom** [comment]
@ricardo oh ok, thank you for the explanation and going out of your way to help, I do appreciate you! Have a nice holiday season!

**Comment by Hastibe** [comment]
I came from [this support thread](https://wordpress.org/support/topic/static-icons-not-showing-on-5-6-2/page/2/), after [opening my own](https://wordpress.org/support/topic/missing-credit-card-images-on-checkout-page/), wondering why the credit card brand images had been removed from displaying at checkout, and I notice that the response was that "This issue is still open and being discussed on GitHub" (here).

I was wondering, what is being discussed about this? I don't see anyone who doesn't want the static credit card brand logos to be returned to the checkout page, or any issues raised about it. ...If there is no opposition to it, and a lot of interest in returning them, could they be added back?

**Comment by iguanamom** [comment]
@Hastibe I just updated to that beta version ("early access") which shows the cards instead. Worked for me. They said they were eventually moving to that version of it anyway so it will be a non-issue. They pointed out in the one you're having problems with that once they start to enter the card number, the CC icon that corresponds to their card appears. Not a great solution IMHO as people want to see the cards on there, looks sketchy without SOMETHING. I still can't add in my own language though to that little section which I'd still like to do but it does say 'Credit Card/Debit Card' which is good enough I suppose.

**Comment by diegocurbelo** [comment]
Fixed in #2038
